src/UnoDrive.Shared/ViewModels/MyFilesViewModel.cs

```diff
@@ -126,7 +133,7 @@ async Task OnBackAsync()
 
 			var backId = location.Back.Id;
 			location = location.Back;
-			await LoadDataAsync(backId);
+			await LoadDataAsync(backId).ConfigureAwait(false);
```

Comment:
There are some people that don't like to return a Task, as I suggested on `OnForwardAsync` method, said that it's fine to mark the method as `async`, but for perf win would be good to mark the last method with the `.ConfigureAwait(false)`. This will affect just this method and since is the last line we don't care what thread will run the continuation

More [info here](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task)

Reply:
When we use `ConfigureAwait(false)` it won't return on the same thread afaik. Should we be worrying about `LoadDataAsync` not being on the main UI thread then? It sets the main `FilesAndFolders` and we should ensure that remains on the main thread.

As far as I understand this will still work, but code after line 136 won't be on the main ui thread.

Reply:
Yes, you're correct, there's a huge chance that the code doesn't return to the UI thread, we can't tell which thread will run the continuation. But this is the last line of the method, so don't care what thread will run the continuation, if, after that line of code you have something else, that could be an issue.